lib/dependencies/LocalModulesHelpers.js

```diff
@@ -13,11 +13,32 @@ LocalModulesHelpers.addLocalModule = function(state, name) {
 	return m;
 };
 
-LocalModulesHelpers.getLocalModule = function(state, name) {
+LocalModulesHelpers.getLocalModule = function(state, name, namedModule) {
 	if(!state.localModules) return null;
+	if(namedModule) {
+		// resolve dependency name relative to the defining named module
+		name = lookup(namedModule, name);
+	}
 	for(var i = 0; i < state.localModules.length; i++) {
 		if(state.localModules[i].name === name)
 			return state.localModules[i];
 	}
 	return null;
 };
+
+function lookup(parent, mod) {
```

Comment:
This is basically copied from https://github.com/webpack/webpack/blob/e773f4de1db24ec147e0203de10ae1a33b02ca82/test/browsertest/mocha.js#L32-L48 . Let me know if you prefer this shared as a module, though I assume mocha.js is loaded in the browser.